### ISSUE TITLE
pwntools changes Python3's print function behavior

### ISSUE THREAD

**Issue by amlamarra** [Gallopsled/pwntools#1873]
I'm trying to write a script that prints over a line as a progress indicator. Take this, for example:

```python
import time

for i in range(0x10):
    print(f"Value: {i:#04x}", end="\r")
    time.sleep(0.1)
print()
```
This works as intended. However, as soon as I add `from pwn import *` to the top, the carriage return ending no longer works.
```plaintext
$ python3 test.py 
Value: 0x00Value: 0x01Value: 0x02Value: 0x03Value: 0x04Value: 0x05Value: 0x06Value: 0x07...
```

I'm on Xubuntu 20.04 with pwntools v4.4.0 installed from pip.

**Comment by Arusekk** [comment]
Related #1653 #1728 and probably already solved by #1716.
Try installing the latest beta version and if it solves the problem, please close.

BTW, don't do that. Use [`log.progress`](https://docs.pwntools.com/en/stable/log.html#pwnlib.log.Logger.progress).

**Comment by amlamarra** [comment]
I installed the latest beta version and now the carriage return seems to be working normally... Except for one minor thing. The line gets wiped out after the last value is printed. Without pwntools being imported, the last value is left there. Not a big deal, just wanted you to know.

And now that I know about `log.progress`, I'll be using that from now on, thanks!
